.github/workflows/rhel7.yml

```diff
@@ -0,0 +1,126 @@
+name: RHEL 7 x86_64
+on:
+  schedule:
+    - cron: "0 1 * * *"
+  workflow_dispatch: {}
+env:
+  BTFHUB_NO_SUDO: true
+jobs:
+  rhel7-update:
+    name: rhel 7 x86_64
+    runs-on: ubuntu-latest
+    container:
+      image: registry.access.redhat.com/rhel7:latest
+    steps:
+      - name: setup subscription
+        run: |
+          rm /etc/rhsm-host
+          yum repolist --disablerepo=*
+          subscription-manager register --org=${{ secrets.DD_BTFHUB_RHEL_ORG_ID }} --activationkey="btfhub-ci"
+          subscription-manager repos --enable=rhel-7-server-debug-rpms --enable=rhel-7-server-rpms
+          yum -y update
+
+      - name: install required packages
+        run: |
+          yum install -y yum-utils wget bzip2 zlib-devel m4 xz gzip cmake find make
+
+      - name: install clang 5 and gcc 7
+        run: |
+          subscription-manager repos --enable rhel-7-server-optional-rpms --enable rhel-server-rhscl-7-rpms --enable rhel-7-server-devtools-rpms
+          cd /etc/pki/rpm-gpg
+          wget -O RPM-GPG-KEY-redhat-devel https://www.redhat.com/security/data/a5787476.txt
+          rpm --import RPM-GPG-KEY-redhat-devel
+          yum install -y devtoolset-7 llvm-toolset-7
+          echo "/opt/rh/devtoolset-7/root/usr/bin" >> $GITHUB_PATH
+          echo "/opt/rh/llvm-toolset-7/root/usr/sbin" >> $GITHUB_PATH
+          echo "/opt/rh/llvm-toolset-7/root/usr/bin" >> $GITHUB_PATH
+          subscription-manager repos --disable rhel-7-server-optional-rpms --disable rhel-server-rhscl-7-rpms --disable rhel-7-server-devtools-rpms
+
+      - name: install newer git
+        run: |
+          yum remove -y git
+          yum install -y https://repo.ius.io/ius-release-el7.rpm https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+          yum install -y git236
+          yum-config-manager --add-repo https://cli.github.com/packages/rpm/gh-cli.repo
+          yum install -y gh
+
+      - name: github authenticate
+        run: |
+          gh auth login --with-token <<<'${{ secrets.DD_BTFHUB_BOT_GITHUB_TOKEN }}'
+
+      - name: checkout btfhub
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.DD_BTFHUB_BOT_GITHUB_TOKEN }}
+          submodules: 'recursive'
+
+      - run: git config --system --add safe.directory $(pwd)
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+
+      - name: build elfutils
+        run: |
+          wget -nv https://sourceware.org/elfutils/ftp/0.190/elfutils-0.190.tar.bz2
```

Comment:
is this a trusted source?

Reply:
Yes. Did you look at the website? They are the authors.